src/api/app/mixins/webui/has_flags.rb

```diff
@@ -0,0 +1,48 @@
+module Webui::HasFlags
+  def create_flag
+    authorize main_object, :update?
+
+    @flag = main_object.flags.new( status: params[:status], flag: params[:flag] )
+    @flag.architecture = Architecture.find_by_name(params[:architecture])
+    @flag.repo = params[:repository] unless params[:repository].blank?
+    @blah = 'whatever'
+
+    respond_to do |format|
+      if @flag.save
+        format.js {}
+      else
+        format.json { render json: @flag.errors, status: :unprocessable_entity }
+      end
+    end
+  end
+
+  def toggle_flag
+    authorize main_object, :update?
+
+    @flag = Flag.find(params[:flag])
+    if @flag.status == 'enable'
```

Comment:
In one line:
``
@flag.status = @flag.status == 'enable' ? 'disable' : 'enable'
``

or if you introduce a ``enabled?`` / ``disabled?`` method:

`` @flag.status = @flag.enabled? ? 'disable' : 'enable' ``

But I'd move the whole to the flag model in a ``toogle`` (or ``toogle!``) function!